src/Features/Core/Portable/CodeCleanup/AbstractCodeCleanupService.cs

```diff
@@ -136,7 +156,92 @@ protected AbstractCodeCleanupService(ICodeFixService codeFixService)
             return document;
         }
 
+        private async Task<Document> ApplyThirdPartyCodeFixesAsync(Document document, IProgressTracker progressTracker, CodeActionOptions options, CancellationToken cancellationToken)
+        {
+            var tree = await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var range = new TextSpan(0, tree.Length);
+
+            // Compute diagnostics for everything that is not an IDE analyzer
+            var diagnostics = (await _diagnosticService.GetDiagnosticsForSpanAsync(document, range,
+                shouldIncludeDiagnostic: static diagnosticId => !(IDEDiagnosticIdToOptionMappingHelper.IsKnownIDEDiagnosticId(diagnosticId) || IsCompilerDiagnostic(diagnosticId)),
```

Comment:
```suggestion
                shouldIncludeDiagnostic: static diagnosticId => !(IDEDiagnosticIdToOptionMappingHelper.IsKnownIDEDiagnosticId(diagnosticId)),
```

Compiler diagnostics should not be excluded here (third party packages can provide fixes for compiler warnings).

Reply:
I could be argued in this direction. I feel like compiler warnings are not in the spirit of how I envisioned this feature but if you can think of a use case I wouldn't mind the change.

Reply:
how does the IDEDiagnosticIdToOptionMappingHelper work?  how does it know what are IDE diagnostics and what aren't?

Reply:
IDEDiagnosticIdToOptionMappingHelper is what all of our code style analyzers register with on their creation. Its the common way we identify internal code style analyzers.